agent/ui_endpoint_test.go

```diff
@@ -103,20 +134,30 @@ func TestUiNodes(t *testing.T) {
 	}
 	assertIndex(t, resp)
 
-	// Should be 2 nodes, and all the empty lists should be non-nil
+	// Should be 3 nodes, and all the empty lists should be non-nil
 	nodes := obj.(structs.NodeDump)
-	if len(nodes) != 2 ||
-		nodes[0].Node != a.Config.NodeName ||
-		nodes[0].Services == nil || len(nodes[0].Services) != 1 ||
-		nodes[0].Checks == nil || len(nodes[0].Checks) != 1 ||
-		nodes[1].Node != "test" ||
-		nodes[1].Services == nil || len(nodes[1].Services) != 0 ||
-		nodes[1].Checks == nil || len(nodes[1].Checks) != 0 {
-		t.Fatalf("bad: %v", obj)
-	}
+	require.Equal(t, 3, len(nodes))
+	require.Equal(t, a.Config.NodeName, nodes[0].Node)
+	require.NotNil(t, nodes[0].Services)
+	require.Equal(t, 1, len(nodes[0].Services))
+	require.NotNil(t, nodes[0].Checks)
+	require.Equal(t, 1, len(nodes[0].Checks))
+	require.Equal(t, "test", nodes[1].Node)
+	require.NotNil(t, nodes[1].Services)
+	require.Equal(t, 0, len(nodes[1].Services))
+	require.NotNil(t, nodes[1].Checks)
+	require.Equal(t, 0, len(nodes[1].Checks))
```

Comment:
This block is hard to read; can we make the `len` checks `require.Len` and maybe add some comments to break up the chunks of assertions?